sdk/eventhubs/azure_messaging_eventhubs/README.md

```diff
@@ -162,47 +162,42 @@ Examples for various scenarios can be found on in the samples directory in our G
 ## Open an Event Hubs message producer on an Event Hubs instance.
 
 ```rust no_run
-use azure_messaging_eventhubs::{ProducerClient, ProducerClientOptions};
+use azure_messaging_eventhubs::ProducerClient;
 
-async fn open_producer_client() -> Result<ProducerClient,azure_core::Error>
-{
+async fn open_producer_client() -> Result<ProducerClient, azure_core::Error> {
     let host = "<EVENTHUBS_HOST>";
     let eventhub = "<EVENTHUB_NAME>";
 
     let credential = azure_identity::DefaultAzureCredential::new()?;
 
-    let producer = azure_messaging_eventhubs::ProducerClient::new(
-        host.to_string(),
-        eventhub.to_string(),
-        credential.clone(),
-        None,
-        );
-    producer.open().await?;
+    let producer =
+        azure_messaging_eventhubs::ProducerClient::builder(host, eventhub, credential.clone())
```

Comment:
Why are you passing parameters to a builder? Doesn't seem very builder-y.

Reply:
QQ: How do you express required parameters in a builder? I figured that making them be parameters on the creation of the builder made the most sense.

I guess I could put them as parameters on the terminal `open()` call instead.

But fundamentally, they need to be *somewhere*.

Reply:
Moved the parameters to the `open` method. The more that I think about it, it makes sense for them to be there, since most of them have to do with the network connection operation anyway.

Reply:
There's also the typestate pattern: https://github.com/heaths/typestate-rs. That said, [guidelines](https://azure.github.io/azure-sdk/rust_introduction.html#rust-builders) don't currently prescribe this scenario. We could say that required parameters for builders are passed as "constructor" args for the builder. I'm not opposed to it, but it was confusing since you have `Option<T>` "required" args.